Framework/DataObjects/src/EventList.cpp

```diff
@@ -1454,6 +1454,15 @@ const MantidVec &EventList::dataE() const {
   return sharedE()->rawData();
 }
 
+namespace {
+inline double calcNorm(const double errorSquared) {
+  if (errorSquared == 0. || errorSquared == 1.)
```

Comment:
Is it better to return a weight of `0` when `errorSquared=0`?

Reply:
`0` is probably better so it can be distinguished from the `1` case.